lib/adal-angular.js

```diff
@@ -175,16 +185,25 @@ if (typeof module !== 'undefined' && module.exports) {
                 var routeChangeHandler = function (e, nextRoute) {
                     if (nextRoute && nextRoute.$$route && isADLoginRequired(nextRoute.$$route, _adal.config)) {
                         if (!_oauthData.isAuthenticated) {
-                            _adal.info('Route change event for:' + $location.$$path);
+                            _adal.info('Route change event for:' + $location.$$url);
                             loginHandler();
                         }
                     }
                 };
 
-                var stateChangeHandler = function (e, nextRoute) {
-                    if (nextRoute && isADLoginRequired(nextRoute, _adal.config)) {
+                var stateChangeHandler = function (e, toState, toParams, fromState, fromParams){
```

Comment:
```
/** @ngInject */
export function routerConfig($stateProvider: ng.ui.IStateProvider, $urlRouterProvider: ng.ui.IUrlRouterProvider) {
    $stateProvider
		.state('home', {
			url: '/',
            views: {
                '': {
                    templateUrl: 'app/home/home.view.html',
                    controller: 'HomeController',
                    controllerAs: 'home'
                }
			}
        })
        .state('entities', {
            abstract: true, // Set this to true so it will never be activated directly
            url: '',
            templateUrl: 'app/main/coraMain.view.html',
            controller: 'CoraMainController',
            controllerAs: 'main'
        })
        .state('secret', <any>{
            url: '/secret?param1&param2&param3',
            templateUrl: 'app/secret.view.html',
            controller: 'SecretController',
            controllerAs: 'secret',
            requireADLogin: true
        });

	$urlRouterProvider.otherwise('/');
}
```